modules/wallet/transactionbuilder.go

```diff
@@ -555,7 +558,10 @@ func (tb *transactionBuilder) Sign(wholeTransaction bool) ([]types.Transaction,
 	}
 	for _, inputIndex := range tb.siafundInputs {
 		input := tb.transaction.SiafundInputs[inputIndex]
-		key := tb.wallet.keys[input.UnlockConditions.UnlockHash()]
+		key, ok := tb.wallet.keys[input.UnlockConditions.UnlockHash()]
+		if !ok {
+			return nil, errors.New("transaction builder added an input that it cannot sign")
```

Comment:
Is this a codepath that you have reached? I agree with the code being here, just wondering if you added it because of issues, or if you're just programming defensively.